wrap/src/hpx_wrap.cpp

```diff
@@ -17,14 +18,27 @@ namespace hpx_start {
     // include_libhpx_wrap is a weak symbol which helps to determine the course
     // of function calls at runtime. It has a default value of `false` which
     // corresponds to the program's entry point being main().
-    // It is overridden in hpx/hpx_main.hpp. Thus, inclusion of the header file
+    // It is overridden in hpx/hpx_main.hpp or set as TRUE if HPX_AUTO_WRAP_MAIN_ACTIVATE is defined.
+    // Thus, inclusion of the header file or defining HPX_AUTO_WRAP_MAIN_ACTIVATE
     // will change the program's entry point to HPX's own custom entry point
     // initialize_main. Subsequent calls before entering main() are handled
     // by this code.
     HPX_SYMBOL_EXPORT extern bool include_libhpx_wrap;
-    HPX_SYMBOL_EXPORT bool include_libhpx_wrap __attribute__((weak)) = false;
+    HPX_SYMBOL_EXPORT bool include_libhpx_wrap __attribute__((weak)) =
+#if defined(HPX_AUTO_WRAP_MAIN_ACTIVATE)
+        true;
+#else
+        false;
+#endif
```

Comment:
Let's write this as:
```suggestion
#if defined(HPX_AUTO_WRAP_MAIN_ACTIVATE)
    HPX_SYMBOL_EXPORT bool include_libhpx_wrap __attribute__((weak)) = true;
#else
    HPX_SYMBOL_EXPORT bool include_libhpx_wrap __attribute__((weak)) = false;
#endif
```
making it better readable.